src/include/miopen/find_solution.hpp

```diff
@@ -63,8 +63,15 @@ auto FindSolutionImpl(rank<1>,
     static_assert(std::is_invocable_v<Db>,
                   "db is meant to be a functor returning a reference to perfdb");
 
-    const FindEnforce enforce =
-        options && options->find_enforce ? *options->find_enforce : FindEnforce{};
+    const auto enforce = [&]() {
+        if(options && options->find_enforce)
+            return *options->find_enforce;
+        const auto& handle = context.GetStream();
+        if(handle.GetTuningPolicy() != miopenTuningPolicyNone)
+            return FindEnforce{static_cast<FindEnforceAction>(handle.GetTuningPolicy())};
+        return FindEnforce{};
+    }();
```

Comment:
Looks like MIOPEN_FIND_ENFORCE is higher priority than TuningPolicy.

Reply:
TuningPolicy has higher priority

Reply:
My mistake, I misread.